src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,2005 @@
+            Diagnostic(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToOverride, "OverloadResolutionPriority(2)").WithLocation(14, 6)
+        };
+
+        comp.VerifyDiagnostics(expectedErrors);
+        verify(comp);
+
+        var comp2 = CreateCompilation([code, OverloadResolutionPriorityAttributeDefinition]);
+        comp2.VerifyDiagnostics(expectedErrors);
+        comp = CreateCompilation(executable, references: [comp2.ToMetadataReference()]);
+        comp.VerifyDiagnostics();
+        verify(comp);
+
+        static void verify(CSharpCompilation comp)
+        {
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var invocation = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
+
+            var method = (IMethodSymbol)model.GetSymbolInfo(invocation).Symbol!;
+            Assert.Equal("void Derived.M(System.Object o)", method.ToTestDisplayString());
+        }
+    }
+
+    [Fact]
+    public void Overrides_ChangePriorityInMetadata_Methods()
+    {
+        var source1 = """
+            using System.Runtime.CompilerServices;
+
+            public class Base
+            {
+                [OverloadResolutionPriority(1)]
+                public virtual void M(object o) => System.Console.WriteLine("1");
+                public virtual void M(string s) => throw null;
+            }
+            """;
+
+        var comp1 = CreateCompilation([source1, OverloadResolutionPriorityAttributeDefinition], assemblyName: "assembly1");
+        var assembly1 = comp1.EmitToImageReference();
+
+        // Equivalent to:
+        //
+        // public class Derived : Base
+        // {
+        //     [OverloadResolutionPriority(0)]
+        //     public override void M(object o) => System.Console.WriteLine("1");
+        //     [OverloadResolutionPriority(2)]
+        //     public override void M(string s) => throw null;
+        // }
+        var il2 = """
+            .assembly extern assembly1 {}
+
+            .class public auto ansi beforefieldinit Derived
+                extends [assembly1]Base
+            {
+                .custom instance void [mscorlib]System.Runtime.CompilerServices.NullableContextAttribute::.ctor(uint8) = (
+                    01 00 01 00 00
+                )
+                .custom instance void [mscorlib]System.Runtime.CompilerServices.NullableAttribute::.ctor(uint8) = (
+                    01 00 00 00 00
+                )
+                // Methods
+                .method public hidebysig virtual 
+                    instance void M (
+                        object o
+                    ) cil managed 
+                {
+                    .custom instance void [assembly1]System.Runtime.CompilerServices.OverloadResolutionPriorityAttribute::.ctor(int32) = (
+                        01 00 00 00 00 00 00 00
+                    )
+                    // Method begins at RVA 0x2050
```

Comment:
nit: consider cleaning up IL to make the test a bit more compact. Also applies below